diff --git a/src/LoreFileContext.c b/src/LoreFileContext.c
new file mode 100644
index 0000000..83fbf7d
--- /dev/null
+++ b/src/LoreFileContext.c
@@ -0,0 +1,29 @@
+LORE_CCG_IMPL(1)
+
+static struct LoreStaticCallCheckGuardInfo LoreFileContext_CCGs[] = {
+    {"void (*)(void *)", &LoreFileContext_CCG_Tramp_1},
+};
+
+static struct LoreStaticFunctionDecayGuardInfo LoreFileContext_FDGs[] = {
+};
+
+static void *LoreFileContext_initializers[] = {
+};
+
+__LORE_PRIV__ struct LoreFileContext LoreFileContext_instance = {
+    .abiVersion = LORE_HLR_ABI_VERSION,
+    .structSize = sizeof(struct LoreFileContext),
+    .runtimeContext = 0,
+    .emuAddr = 0,
+    .setThreadCallback = 0,
+    .numCCGs = 1,
+    .CCGs = LoreFileContext_CCGs,
+    .numFDGs = 0,
+    .FDGs = LoreFileContext_FDGs,
+    .numInitializers = 0,
+    .initializers = LoreFileContext_initializers,
+};
+
+__LORE_EXPORT__ struct LoreFileContext *LoreGetFileContext() {
+    return &LoreFileContext_instance;
+}
diff --git a/src/LoreFileContext.h b/src/LoreFileContext.h
new file mode 100644
index 0000000..7cd92c2
--- /dev/null
+++ b/src/LoreFileContext.h
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <lorelei/RuntimeExtension/HLRExtension.h>
+
+#define __LORE_PRIV__ __attribute__((visibility("hidden")))
+#define __LORE_EXPORT__ __attribute__((visibility("default")))
+#define __LORE_UNLIKELY__(EXPR) __builtin_expect(!!(EXPR), 0)
+#define __LORE_OFFSET_OF__(TYPE, MEMBER) __builtin_offsetof(TYPE, MEMBER)
+#define __LORE_FUNC_TYPE__(EXPR) __typeof__(&*(EXPR))
+#define __LORE_SIZE_T__ uintptr_t
+
+#define LoreFunctionTrampoline LoreFunctionTrampolineLayout
+#define LoreStaticCallCheckGuardInfo LoreHLRCallCheckGuardInfo
+#define LoreStaticFunctionDecayGuardPair LoreHLRFunctionDecayGuardPair
+#define LoreStaticFunctionDecayGuardInfo LoreHLRFunctionDecayGuardInfo
+#define LoreRuntimeContext LoreHLRRuntimeContext
+#define LoreFileContext LoreHLRFileContext
+
+#define __LORE_MAGIC_SIGN__ LORE_FUNCTION_TRAMPOLINE_MAGIC
+#define __LORE_THUNK_INSTR_OFFSET__ \
+    __LORE_OFFSET_OF__(struct LoreFunctionTrampoline, thunk_instr)
+
+extern __LORE_PRIV__ struct LoreFileContext LoreFileContext_instance;
+
+static inline void *__LoreFileContext_CCG(void *addr, void *tramp) {
+    if (__LORE_UNLIKELY__((__LORE_SIZE_T__) addr <
+                          (__LORE_SIZE_T__) LoreFileContext_instance.emuAddr)) {
+        struct LoreFunctionTrampoline *maybeGuestTramp =
+            (struct LoreFunctionTrampoline *) ((char *) addr - __LORE_THUNK_INSTR_OFFSET__);
+        if (__LORE_UNLIKELY__(maybeGuestTramp->magic_sign == __LORE_MAGIC_SIGN__)) {
+            addr = maybeGuestTramp->saved_function;
+        } else {
+            LoreFileContext_instance.setThreadCallback(addr);
+            addr = tramp;
+        }
+    }
+    return addr;
+}
+
+#define LORE_CCG_IMPL(NUM) __LORE_PRIV__ void *LoreFileContext_CCG_Tramp_##NUM = 0;
+#define LORE_CCG_GET(NUM, EXPR) \
+    ((__LORE_FUNC_TYPE__(EXPR)) __LoreFileContext_CCG((void *) (EXPR), \
+                                                      LoreFileContext_CCG_Tramp_##NUM))
+
+extern __LORE_PRIV__ void *LoreFileContext_CCG_Tramp_1;
+
+#define LORE_CCG_1(X) LORE_CCG_GET(1, (X))
diff --git a/src/xcb_list.c b/src/xcb_list.c
index bdd2d43..0ba339c 100644
--- a/src/xcb_list.c
+++ b/src/xcb_list.c
@@ -33,6 +33,7 @@
 
 #include "xcb.h"
 #include "xcbint.h"
+#include "LoreFileContext.h"
 
 typedef struct node {
     struct node *next;
@@ -66,7 +67,7 @@ void _xcb_map_delete(_xcb_map *list, xcb_list_free_func_t do_free)
     {
         node *cur = list->head;
         if(do_free)
-            do_free(cur->data);
+            LORE_CCG_1(do_free)(cur->data);
         list->head = cur->next;
         free(cur);
     }
diff --git a/src/xcb_out.c b/src/xcb_out.c
index a4fe823..6cab1bb 100644
--- a/src/xcb_out.c
+++ b/src/xcb_out.c
@@ -41,6 +41,7 @@
 #include "xcb.h"
 #include "xcbext.h"
 #include "xcbint.h"
+#include "LoreFileContext.h"
 #include "bigreq.h"
 
 static inline void send_request(xcb_connection_t *c, int isvoid, enum workarounds workaround, int flags, struct iovec *vector, int count)
@@ -97,7 +98,7 @@ static void get_socket_back(xcb_connection_t *c)
 
     c->out.socket_moving = 1;
     pthread_mutex_unlock(&c->iolock);
-    c->out.return_socket(c->out.socket_closure);
+    LORE_CCG_1(c->out.return_socket)(c->out.socket_closure);
     pthread_mutex_lock(&c->iolock);
     c->out.socket_moving = 0;
 
@@ -463,6 +464,8 @@ void _xcb_out_destroy(_xcb_out *out)
     pthread_cond_destroy(&out->socket_cond);
 }
 
+#include "LoreFileContext.c"
+
 int _xcb_out_send(xcb_connection_t *c, struct iovec *vector, int count)
 {
     int ret = 1;
