diff --git a/tests/fuzzer.c b/tests/fuzzer.c
index f9e4d20..2e4db95 100644
--- a/tests/fuzzer.c
+++ b/tests/fuzzer.c
@@ -1961,6 +1961,7 @@ static int basicUnitTests(U32 seed, double compressibility)
 
     /* Static CCtx tests */
 #define STATIC_CCTX_LEVEL 4
+    if (getenv("ZSTD_SKIP_STATIC_CONTEXT_TESTS") == NULL) {
     DISPLAYLEVEL(3, "test%3i : create static CCtx for level %u : ", testNb++, STATIC_CCTX_LEVEL);
     {   size_t const staticCStreamSize = ZSTD_estimateCStreamSize(STATIC_CCTX_LEVEL);
         void* const staticCCtxBuffer = malloc(staticCStreamSize);
@@ -2187,6 +2188,7 @@ static int basicUnitTests(U32 seed, double compressibility)
         free(staticCCtxBuffer);
         free(staticDCtxBuffer);
     }
+    }
 
     DISPLAYLEVEL(3, "test%3i : Static context sizes for negative levels : ", testNb++);
     {   size_t const cctxSizeN1 = ZSTD_estimateCCtxSize(-1);
diff --git a/tests/zstreamtest.c b/tests/zstreamtest.c
index 0f76826..42d18bd 100644
--- a/tests/zstreamtest.c
+++ b/tests/zstreamtest.c
@@ -1918,6 +1918,7 @@ static int basicUnitTests(U32 seed, double compressibility)
     }
     DISPLAYLEVEL(3, "OK \n");
 
+    if (getenv("ZSTD_SKIP_SEQUENCE_PRODUCER_TESTS") == NULL) {
     DISPLAYLEVEL(3, "test%3i : Block-Level External Sequence Producer API: ", testNb++);
     {
         size_t const dstBufSize = ZSTD_compressBound(CNBufferSize);
@@ -2060,6 +2061,7 @@ static int basicUnitTests(U32 seed, double compressibility)
         free(checkBuf);
     }
     DISPLAYLEVEL(3, "OK \n");
+    }
 
 
     /* Test maxBlockSize cctx param functionality */
