diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0387f4..4ad550a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -673,5 +673,5 @@
 endif()
 
-if(BUILD_TESTING AND NOT BUILD_SHARED_LIBS)
+if(BUILD_TESTING)
   enable_testing()
 
@@ -721,15 +721,17 @@ if(BUILD_TESTING AND NOT BUILD_SHARED_LIBS)
         -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
         -P "${PROJECT_SOURCE_DIR}/cmake/RunTest.cmake")
 
+  if(NOT BUILD_SHARED_LIBS)
   add_executable(test_opus_extensions ${test_opus_extensions_sources})
   target_include_directories(test_opus_extensions
                             PRIVATE ${CMAKE_CURRENT_BINARY_DIR} celt dnn)
   target_link_libraries(test_opus_extensions PRIVATE opus)
   target_compile_definitions(test_opus_extensions PRIVATE OPUS_BUILD)
   add_test(NAME test_opus_extensions COMMAND ${CMAKE_COMMAND}
         -DTEST_EXECUTABLE=$<TARGET_FILE:test_opus_extensions>
         -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
         -P "${PROJECT_SOURCE_DIR}/cmake/RunTest.cmake")
+  endif()
   if(OPUS_DRED)
     add_executable(test_opus_dred ${test_opus_dred_sources})
     target_include_directories(test_opus_dred
